File: Swift/ExampleTests/production/MatchCollection.swift
public let matchCollectionMinimum = 0
public let matchCollectionMaximum = 10000

public class MatchCollection: Equatable {
    private var items: [Match] = []

    public init() {}

    public static func == (a: MatchCollection, b: MatchCollection) -> Bool {
        return a.items == b.items
    }

    public func add(_ item: Match) {
        items.append(item)
    }

    public func delete(_ item: Match) -> Bool {
        if let idx = items.firstIndex(where: { $0 == item }) {
            items.remove(at: idx)
            return true
        }
        return false
    }

    public func read() -> [Match] {
        return items
    }

    public func update(_ oldItem: Match, with newItem: Match) -> Bool {
        if let idx = items.firstIndex(where: { $0 == oldItem }) {
            items[idx] = newItem
            return true
        }
        return false
    }

    public var size: Int { return items.count }
}

File: Rust/ExampleTests/production/matchcollection.rs
use super::*;

pub const MATCHCOLLECTION_MINIMUM: usize = 0;
pub const MATCHCOLLECTION_MAXIMUM: usize = 10000;

#[derive(Debug, Clone, Default, PartialEq)]
pub struct MatchCollection {
    items: Vec<Match>,
}

impl MatchCollection {
    pub fn new() -> Self { Self::default() }

    pub fn add(&mut self, item: Match) {
        self.items.push(item);
    }

    pub fn delete(&mut self, item: &Match) -> bool
    where
        Match: PartialEq,
    {
        if let Some(pos) = self.items.iter().position(|x| x == item) {
            self.items.remove(pos);
            true
        } else { false }
    }

    pub fn read(&self) -> &[Match] {
        &self.items
    }

    pub fn update(&mut self, old_item: &Match, new_item: Match) -> bool
    where
        Match: PartialEq,
    {
        if let Some(pos) = self.items.iter().position(|x| x == old_item) {
            self.items[pos] = new_item;
            true
        } else { false }
    }

    pub fn size(&self) -> usize { self.items.len() }
}
